### PR TITLE
Fix test

### LINKED ISSUE #1: Fix ClosureTestwiseCoverageGeneratorTest test
```
eu.cqse.teamscale.report.testwise.closure.ClosureTestwiseCoverageGeneratorTest > testTestwiseReportGeneration FAILED
    junit.framework.ComparisonFailure at ClosureTestwiseCoverageGeneratorTest.java:25

```

### PR DESCRIPTION
**Pull request by DreierF**
- [x] [No new Teamscale findings](https://build.cqse.eu/teamscale/delta.html#input/jacoco-client/?showMergeFindings=true)
- [x] [Changes are tested adequately](https://build.cqse.eu/teamscale/tests.html#/jacoco-client/)
- [x] USAGE.md updated in case of user-visible changes
- [x] CHANGELOG.md updated
- [x] build.gradle version number increased according to semantic versioning

Fixed #1

### CODE REVIEW

**Comment by DreierF** [comment]
The problem was that the order of the xml elements in the report was non-deterministic. the entries are now sorted in lexicographical order

File: report-generator/src/main/java/eu/cqse/teamscale/report/testwise/model/FileCoverage.java [hunk 1]
```diff
@@ -107,6 +107,11 @@ public boolean isEmpty() {
 		return coveredRanges.isEmpty();
 	}
 
+	@Override
```
Comment (karottenreibe) [inline report-generator/src/main/java/eu/cqse/teamscale/report/testwise/model/FileCoverage.java]: I don't like comparables that leave out part of the attributes. To me it's unexpected and s subtle source of bugs. I'd prefer a comparator. That way you can clarify what is being compared with the comparators name.

File: report-generator/src/main/java/eu/cqse/teamscale/report/testwise/model/PathCoverage.java [hunk 2]
```diff
@@ -37,6 +39,11 @@ public void add(FileCoverage fileCoverage) {
 	/** Returns a collection of {@link FileCoverage}s associated with this path. */
 	@XmlElement(name = "file")
 	public Collection<FileCoverage> getFiles() {
-		return fileCoverageList.values();
+		return CollectionUtils.sort(fileCoverageList.values());
+	}
+
+	@Override
+	public int compareTo(PathCoverage o) {
```
Comment (karottenreibe) [inline report-generator/src/main/java/eu/cqse/teamscale/report/testwise/model/PathCoverage.java]: See my comments for the other compare to method

File: report-generator/src/main/java/eu/cqse/teamscale/report/testwise/model/TestCoverage.java [hunk 3]
```diff
@@ -50,11 +52,16 @@ public void addAll(List<FileCoverage> fileCoverageList) {
...
 
 	/** Returns true if there is no coverage for the test yet. */
 	public boolean isEmpty() {
 		return pathCoverageList.isEmpty();
 	}
+
+	@Override
+	public int compareTo(TestCoverage o) {
+		return this.externalId.compareTo(o.externalId);
+	}
```
Comment (karottenreibe) [inline report-generator/src/main/java/eu/cqse/teamscale/report/testwise/model/TestCoverage.java]: See my comment for the other compare to method

**Pull request merged**